handlers.go

```diff
@@ -76,22 +80,24 @@ func (h ContentHandler) getTransformedContent(ctx context.Context, nativeContent
 	return h.handleResponse(req, resp, err, w, uuid, h.serviceConfig.transformAppName)
 }
 
-func (h ContentHandler) handleResponse(req *http.Request, extResp *http.Response, err error, w http.ResponseWriter, uuid, calledServiceName string) (bool, *http.Response) {
+func (h ContentHandler) handleResponse(req *http.Request, extResp *http.Response, err error, w http.ResponseWriter, uuid, calledServiceName string) (*http.Response, error) {
 	//this happens when hostname cannot be resolved or host is not accessible
 	if err != nil {
 		h.handleError(w, err, calledServiceName, req.URL.String(), req.Header.Get(tid.TransactionIDHeader), uuid)
-		return false, nil
+		return nil, err
 	}
 	switch extResp.StatusCode {
 	case http.StatusOK:
 		h.log.ResponseEvent(calledServiceName, req.URL.String(), extResp, uuid)
-		return true, extResp
+		return extResp, nil
+	case http.StatusUnprocessableEntity:
+		fallthrough
 	case http.StatusNotFound:
-		h.handleNotFound(w, extResp, calledServiceName, req.URL.String(), uuid)
-		return false, nil
+		h.handleClientError(w, calledServiceName, req.URL.String(), extResp, uuid)
+		return nil, errors.New("not found")
```

Comment:
We need to return the resp itself again instead of nil since there is no `err` and resp.Body needs to be closed.